cmd/restic/cmd_self_update.go

```diff
@@ -21,6 +21,8 @@ GitHub and replaces the currently running binary. After download, the
 authenticity of the binary is verified using the GPG signature on the release
 files.
 
+This command might be disabled, for example in official OS/distribution packages.
```

Comment:
The help output here is intended for the end user. There's no point in adding developer-specific information. Thus, please remove that remark again.

Reply:
I added this precisely for end users, like me.

Never having used restic before, I open restic(1), read about self-updates and try it out.
It works (just checks, already latest version), but I wonder why since this is a system owned executable.

Then yes, as a package maintainer, I dig deeper and learn that this feature should not be enabled in the first place.
It is getting fixed, so now I imagine end users installing the fixed package for the first time.

Being an end user again, I read the manual, see a listing of all commands, matching manual cross-references,
except this time there is one reference too many.

No short command listing, `restic self-update` throws an error, but restic-self-update(1) exists.
Now I'm confused:  is the OS package broken?  Did I misunderstood the tool/its documentation?
Do I have to installer another package to get self-update functionality?

Reading this note, as end user, would certainly make me stop wondering and just expect it to be disabled, rather
than wondering what went wrong on which end.

Reply:
Alternatively, the restic-self-update(1) could be skipped during installation and its reference in restic(1) could be omitted if the `selfupdate` tag is missing, but I'd have to look into how restic generates its docs.

Reply:
> Alternatively, the restic-self-update(1) could be skipped during installation and its reference in restic(1) could be omitted if the `selfupdate` tag is missing, but I'd have to look into how restic generates its docs.

Great, it already does that by default and I just had to rerun restic-generate(1) with this PR.
I removed this useless commit.